iroha_p2p/src/network.rs

```diff
@@ -246,27 +375,33 @@ where
         };
         match Peer::new_from(id.clone(), stream, self.broker.clone()) {
             Ok(peer) => {
+                let connection_id = peer.connection_id;
                 let peer = peer.start().await;
-                self.peers.insert(id, peer);
+                self.new_peers.insert(connection_id, peer.clone());
+                peer.do_send(Connect).await;
             }
             Err(e) => warn!(%e, "Unable to create peer"),
         }
     }
 }
 
-/// The message that is sent to [Network] to start connection to some other peer.
+/// The message that is sent to [`NetworkBase`] to start connection to some other peer.
 #[derive(Clone, Debug, iroha_actor::Message)]
-pub struct Connect {
+pub struct ConnectPeer {
     /// Peer identification
     pub id: PeerId,
 }
 
-/// The message that is sent to [`Network`] to get connected peers' ids.
+/// The message that is sent to [`Peer`] to start connection.
+#[derive(Clone, Copy, Debug, iroha_actor::Message)]
+pub struct Connect;
```

Comment:
Could you show there it is used?

Reply:
`impl<T, K, E> ContextHandler<Connect> for Peer<T, K, E>`
It is used to signal the peer to start the work _after_ the `Actor::on_start()`.